faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -330,23 +336,21 @@ private Value handleResponse(FullHttpResponse response) {
       return json.treeToValue(resource, Value.class);
     } catch (IOException ex) {
       throw new AssertionError(ex);
-    } finally {
-      response.release();
     }
   }
 
   private CompletableFuture<Value> performRequest(JsonNode body, Optional<Duration> queryTimeout) {
     try {
         return handleNetworkExceptions(connection.post("", body, queryTimeout).thenApply(this::handleResponse));
-    } catch (IOException ex) {
+    } catch (Exception ex) {
```

Comment:
This catch is too broad.

Reply:
Yes, it seems we should be returning a `CompletableFuture` (with no checked exceptions) from `FaunaClient.performRequest` in order to avoid all of this.

Reply:
I have backported the improvements from the streaming branch.
https://github.com/fauna/faunadb-jvm/pull/270/commits/b891a23fc2abb953b2bdbc58f3020aa676a9692b

It introduces a request builder that throws less broad exceptions and fails `Future` accordingly.

Reply:
and the final cleanup at the call site https://github.com/fauna/faunadb-jvm/pull/270/commits/323bdf8761193f03c29d321879d440e2f5211815